### PR TITLE
fix: updating peerDependency fields for yarn berry

### PR DESCRIPTION
**Pull request by vidarc**
## Motivation

Updating peerDependency fields to resolve all the warnings that the latest version of yarn outputs.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

👍 

## Test Plan

The following addition to my `.yarnrc.yml` file resolves all the warnings.

```yaml

packageExtensions:
  "@docusaurus/mdx-loader@*":
    peerDependencies:
      "@babel/core": "*"
  "@docusaurus/plugin-content-blog@*":
    peerDependencies:
      "@babel/core": "*"
  "@docusaurus/plugin-content-docs@*":
    peerDependencies:
      "@babel/core": "*"
  "@docusaurus/plugin-content-pages@*":
    peerDependencies:
      "@babel/core": "*"
  "@docusaurus/plugin-debug@*":
    peerDependencies:
      "@babel/core": "*"
      webpack: "*"
  "@docusaurus/plugin-google-analytics@*":
    peerDependencies:
      "@babel/core": "*"
      webpack: "*"
  "@docusaurus/plugin-google-gtag@*":
    peerDependencies:
      "@babel/core": "*"
      webpack: "*"
  "@docusaurus/plugin-sitemap@*":
    peerDependencies:
      "@babel/core": "*"
      webpack: "*"
  "@docusaurus/preset-classic@*":
    peerDependencies:
      "@babel/core": "*"
      prism-react-renderer: "*"
      webpack: "*"
  "@docusaurus/theme-classic@*":
    peerDependencies:
      "@babel/core": "*"
      webpack: "*"
  "@docusaurus/theme-common@*":
    peerDependencies:
      "@babel/core": "*"
  "@docusaurus/theme-search-algolia@*":
    peerDependencies:
      "@babel/core": "*"
      prism-react-renderer: "*"
      webpack: "*"
  "@docusaurus/utils-validation@*":
    peerDependencies:
      "@babel/core": "*"
      webpack: "*"
```

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: packages/docusaurus-utils/package.json [hunk 1]
```diff
@@ -51,6 +51,6 @@
     "@babel/core": "^7.0.0",
     "react": "*",
     "react-dom": "*",
-    "webpack": "5.x"
+    "webpack": "^5.0.0"
```
Comment (Josh-Cena) [inline packages/docusaurus-utils/package.json:54]: `5.x` is the same as `^5.0.0` as I understand?

Comment (vidarc) [reply packages/docusaurus-utils/package.json:54]: I wasn't sure. Was thinking better to standardize the version style. That was the only one using an `x`

Comment (Josh-Cena) [reply packages/docusaurus-utils/package.json:54]: Ah, sure

**Review by Josh-Cena — APPROVED** [review]
Thanks!

**Pull request merged**